src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -193,7 +209,30 @@ def untar_strip_leading_slash(tarfname, path):
 
                 untar_strip_leading_slash(fname, "resources")
                 os.unlink(fname)
-                deps_to_remove.append(dep)
+                # add dep name to deps_downloaded set
+                deps_downloaded.add(dep.get("name"))
+                
+            except DXError:
+                print("Failed to download {} from region {}.".format(handler_id, source_region),
+                        file=sys.stderr)
+                # clean up deps already downloaded and quit downloading
+                deps_downloaded.clear()
+                shutil.rmtree("resources")
+                break
+        # if all deps have been checked without an error, mark downloading as completed
+        else: # for loop finished with no break
+            download_completed = True                    
+    
+    # Check if downloading is completed in one of the enabled regions
+    # if so, files in deps_downloaded will not shown in dxapp.json
+    # if not, deps_downloaded is an empty set. So ID of all deps will be in dxapp.json
+    if not download_completed:
+        print("Downloading resources from region {} failed. "
+              "Please try downloading with their IDs in dxapp.json, "
+              "or skip downloading resources entirely by using the --omit-resources flag.".format(source_region))
+    # if anything has been downloaded
+    if deps_downloaded:
```

Comment:
Have you tried `dx getting` a couple of our public apps with resources/ or bundledDepends? Do they have a lot of items in `deps_downloaded`? Are they familiar or some names of libraries that don't add to the understanding of what was downloaded? If they don't, I'd remove this new message entirely.

Reply:
Yes I have tried the ones that has bundled and asset depends (SAK for example) and most of them have one or two (except for canu, which has asset files without "AssetBundle" that link it to its record)... I will remove it since we already print out that info. Thanks :D